<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Integrative analysis of proteome-scale protein-protein interaction networks from the BioPlex project • BiocChallenges</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<meta property="og:title" content="Integrative analysis of proteome-scale protein-protein interaction networks from the BioPlex project">
<meta property="og:description" content="BiocChallenges">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BiocChallenges</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.7001</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../articles/BiocChallenges.html">Get started</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/challenges/EDAMviews.html">Upgrading biocViews with EDAM</a>
    </li>
    <li>
      <a href="../../articles/challenges/biocchallenges_functionality.html">Contribute new functionality to BiocChallenges</a>
    </li>
    <li>
      <a href="../../articles/challenges/bioplex.html">Integrative analysis of proteome-scale protein-protein interaction networks from the BioPlex project</a>
    </li>
    <li>
      <a href="../../articles/challenges/bugsigdb.html">Systematic analysis of published microbial signatures from BugSigDB</a>
    </li>
    <li>
      <a href="../../articles/challenges/build_failure_tracking.html">Facilitating diagnosis of Bioconductor package build failures</a>
    </li>
    <li>
      <a href="../../articles/challenges/instrumentation.html">Instrumentation for genomic computing with Bioconductor</a>
    </li>
    <li>
      <a href="../../articles/challenges/isee_transcripts.html">Unlocking iSEE for transcript-level visualization</a>
    </li>
    <li>
      <a href="../../articles/challenges/single_cell_gallery.html">Contribute to the SingleCellExperiment Gallery</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../articles/index.html">More...</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kevinrue/BiocChallenges/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="bioplex_files/header-attrs-2.11/header-attrs.js"></script><link href="bioplex_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="bioplex_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Integrative analysis of proteome-scale protein-protein interaction networks from the BioPlex project</h1>
                        <h4 class="author">Ludwig Geistlinger</h4>
            <address class="author_afil">
      Center for Computational Biomedicine, Harvard Medical School<br><a class="author_email" href="mailto:#"></a><a href="mailto:ludwig_geistlinger@hms.harvard.edu" class="email">ludwig_geistlinger@hms.harvard.edu</a>
      </address>
                              <h4 class="author">Robert Gentleman</h4>
            <address class="author_afil">
      Center for Computational Biomedicine, Harvard Medical School<br><small class="dont-index">Source: <a href="https://github.com/kevinrue/BiocChallenges/blob/master/vignettes/challenges/bioplex.Rmd"><code>vignettes/challenges/bioplex.Rmd</code></a></small>
      <div class="hidden name"><code>bioplex.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The <a href="https://bioplex.hms.harvard.edu">BioPlex project</a> has created two proteome-scale, cell-line-specific protein-protein interaction (PPI) networks: the first in 293T cells, including 120k interactions among 15k proteins; and the second in HCT116 cells, including 70k interactions between 10k proteins. The BioPlex R package (<a href="https://github.com/ccb-hms/BioPlex" class="uri">https://github.com/ccb-hms/BioPlex</a>, submitted to Bioconductor) implements access to the BioPlex PPI networks and related resources from within R. Besides PPI networks for 293T and HCT116 cells, this includes access to CORUM protein complex data, and transcriptome and proteome data for the two cell lines. The goal of this challenge is to introduce the BioPlex data and package to the community, and work together on several analysis and programming challenges around the data including: (a) transcriptomic and proteomic data integration on the BioPlex networks, (b) assessing the impact of alternative splicing on bait proteins of the networks, (c) integration with public databases for disease-associated genes and variants, (d) implementing a GraphFrames backend for efficient representation and analysis of the networks, and (e) designing an R/Shiny graph viewer that allows flexible inspection of experimental data and metadata for nodes and edges of the networks.</p>
    </div>
    
<div id="difficulty" class="section level2">
<h2 class="hasAnchor">
<a href="#difficulty" class="anchor"></a>Difficulty</h2>
<p>Advanced.</p>
</div>
<div id="challenge-leaders" class="section level2">
<h2 class="hasAnchor">
<a href="#challenge-leaders" class="anchor"></a>Challenge leader(s)</h2>
<ul>
<li>Ludwig Geistlinger - <i class="fab fa-github"></i> <a href="https://github.com/lgeistlinger">lgeistlinger</a>
</li>
<li>Robert Gentleman - <i class="fab fa-github"></i> <a href="https://github.com/rgentlem">rgentlem</a>
</li>
</ul>
</div>
<div id="date-added" class="section level2">
<h2 class="hasAnchor">
<a href="#date-added" class="anchor"></a>Date added</h2>
<p>This challenge was created on August 04, 2021</p>
</div>
<div id="topics" class="section level2">
<h2 class="hasAnchor">
<a href="#topics" class="anchor"></a>Topics</h2>
<ul>
<li><code>BioPlex</code></li>
</ul>
</div>
<div id="duration" class="section level2">
<h2 class="hasAnchor">
<a href="#duration" class="anchor"></a>Duration</h2>
<p>A minimum of TBD is recommended for this challenge.</p>
</div>
<div id="repositories" class="section level2">
<h2 class="hasAnchor">
<a href="#repositories" class="anchor"></a>Repositories</h2>
<p>A list of GitHub repositories that participate in this challenge.</p>
<pre><code>## Warning in .cache_github_repo(repo): Querying github

## Warning in .cache_github_repo(repo): Querying github

## Warning in .cache_github_repo(repo): Querying github</code></pre>
<table class="table">
<thead><tr class="header">
<th align="left">repository</th>
<th align="right">stargazers</th>
<th align="left">latest.push</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">
<i class="fab fa-github"></i> <a href="https://github.com/ccb-hms/BioPlex">ccb-hms/BioPlex</a>
</td>
<td align="right">3</td>
<td align="left">2021-10-29 16:42:10</td>
</tr>
<tr class="even">
<td align="left">
<i class="fab fa-github"></i> <a href="https://github.com/ccb-hms/BioPlexAnalysis">ccb-hms/BioPlexAnalysis</a>
</td>
<td align="right">0</td>
<td align="left">2021-09-20 23:29:08</td>
</tr>
<tr class="odd">
<td align="left">
<i class="fab fa-github"></i> <a href="https://github.com/ccb-hms/GraphViewer">ccb-hms/GraphViewer</a>
</td>
<td align="right">0</td>
<td align="left">2021-09-28 17:13:32</td>
</tr>
</tbody>
</table>
</div>
<div id="project-1-transcriptomeproteome-data-integration" class="section level2">
<h2 class="hasAnchor">
<a href="#project-1-transcriptomeproteome-data-integration" class="anchor"></a>Project 1: Transcriptome/proteome data integration</h2>
<p><strong>Background:</strong> The BioPlex project has created two proteome-scale, cell-line-specific PPI networks: one for human embryonic kidney <a href="https://en.wikipedia.org/wiki/293T">293T</a> cells, and one for human colon cancer <a href="https://en.wikipedia.org/wiki/HCT116_cells">HCT116</a> cells. The <a href="https://github.com/ccb-hms/BioPlex">BioPlex R package</a> serves transcriptome and proteome data for the two cell lines in dedicated Bioconductor data structures. This includes (i) transcriptome and proteome data for HCT116 from the <a href="https://portals.broadinstitute.org/ccle">Cancer Cell Line Encyclopedia (CCLE)</a> , and (ii) transcriptome and proteome data profiling differential expression between 293T and HCT116 cells from different sources.</p>
<p><strong>Objective:</strong> Can we identify patterns driving (dis-)agreement between transcriptome and proteome of HEK293 and HCT116 cells?</p>
<p><strong>Proposed methods:</strong> The <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/TranscriptomeProteome.html">Transcriptome-Proteome analysis vignette</a> implements a scaffold for integrative analysis of transcriptome and proteome data from both cell lines. This includes demonstration of how to obtain the corresponding datasets and initial results from (i) correlating HCT116 transcript levels and protein levels from CCLE, and (ii) correlating differential gene expression and differential protein abundance when comparing 293T against HCT116 cells.</p>
<p>Given that HCT116 is a cancer cell line, an important driver of transcriptomic and proteomic expression could be somatic copy number alteration. One approach to this problem would thus be to obtain copy number data from <a href="https://www.cbioportal.org">cBioPortal</a> for HCT116 (the <a href="https://bioconductor.org/packages/cBioPortalData">cBioPortalData</a> package provides an R/Bioconductor interface for accessing cBioPortal study data incl. CCLE). A simplified version of this would be to obtain pre-computed <a href="https://maayanlab.cloud/Harmonizome/gene_set/HCT116/CCLE+Cell+Line+Gene+CNV+Profiles">gene sets of high and low copy number in HCT116 from Enrichr</a> and test for their enrichment in genes/proteins that have high or low expression in HCT116 cells (i) individually, and (ii) when compared to 293T cells.</p>
<p>More generally, standard gene set enrichment analysis of eg. GO terms and KEGG pathways could be employed to identify themes in the sets of genes showing agreement for (i) transcript levels and protein expression in HCT116 CCLE samples, and (ii) differential gene and protein expression between 293T and HCT116 cells from different sources.</p>
<p><strong>What a successful result would look like:</strong> Pull request to the <a href="https://github.com/ccb-hms/BioPlexAnalysis">BioPlexAnalysis</a> github repository that extends the <a href="https://ccb-hms.github.io/BioPlexAnalysis/articles/TranscriptomeProteome.html">Transcriptome-Proteome analysis vignette</a>. Pull requests will be reviewed and discussed. Accepted contributions will be acknowledged.</p>
<p><strong>Potential follow-up work</strong>: Analyze PPI networks for differential wiring in areas of differential expression between 293T and HCT116 cells.</p>
</div>
<div id="project-2-network-propagation-of-genetic-disease-associations" class="section level2">
<h2 class="hasAnchor">
<a href="#project-2-network-propagation-of-genetic-disease-associations" class="anchor"></a>Project 2: Network propagation of genetic disease associations</h2>
<p><strong>Background:</strong> Network propagation of genetic disease associations is an approach for identifying potential drug targets (<a href="https://doi.org/10.1038/s41598-020-77847-9">MacNamara et al., 2020</a>). Such approaches typically leverage publicly available collections of genes and variants associated with human diseases such as <a href="https://www.opentargets.org/">OpenTargets</a>. OpenTargets integrates data from expert curated repositories, GWAS catalogues, animal models, and the scientific literature. It has been shown that protein networks from specific functional linkages such as protein complexes are suitable for simple guilt-by-association network propagation approaches (<a href="https://doi.org/10.1038/s41598-020-77847-9">MacNamara et al., 2020</a>). Global protein–protein interaction networks typically require more sophisticated approaches such as <a href="https://github.com/raphael-group/hotnet2">HotNet2</a>.</p>
<p><strong>Objective:</strong> Application of network propagation methods to CORUM complexes and BioPlex networks using genetic disease association scores from OpenTargets.</p>
<p><strong>Proposed methods</strong>: A list of all OpenTargets datasets is available from the <a href="https://platform.opentargets.org/downloads">Platform Data Downloads</a> page. Example scripts for accessing the data from within R are available <a href="https://platform-docs.opentargets.org/data-access/datasets">here</a>. Following <a href="https://doi.org/10.1038/s41598-020-77847-9">MacNamara et al., 2020</a>, one would first use the scored gene-disease associations from OpenTargets to obtain high‑confidence genetic hits (HCGHs). Proxy gene sets of HCGHs can then be defined by identifying genes that (i) share a protein complex with a HCGH, (ii) are annotated to the same KEGG or REACTOME pathway, (iii) are first- or second-degree neighbors in a BioPlex network, and (iv) are found in a network module identified with approaches such as <a href="https://bioconductor.org/packages/BioNet">BioNet</a> or <a href="https://github.com/raphael-group/hotnet2">HotNet2</a>.</p>
<p><strong>What a successful result would look like:</strong> Pull request to the <a href="https://github.com/ccb-hms/BioPlexAnalysis">BioPlexAnalysis</a> github repository outlining the approach in a separate network propagation vignette (<code>.Rmd</code> file in the <code>vignettes</code> folder). Pull requests will be reviewed and discussed. Accepted contributions will be acknowledged.</p>
<p><strong>Potential follow-up work</strong>: Discussion of how to best encapsulate the OpenTargets gene-disease associations in a package with a <a href="https://cran.r-project.org/web/packages/sparklyr/index.html">sparklyr</a> backend.</p>
</div>
<div id="project-3-graphframes-backend-for-representation-of-bioplex-ppi-networks" class="section level2">
<h2 class="hasAnchor">
<a href="#project-3-graphframes-backend-for-representation-of-bioplex-ppi-networks" class="anchor"></a>Project 3: GraphFrames backend for representation of BioPlex PPI networks</h2>
<p><strong>Background:</strong> <a href="https://spark.rstudio.com/graphframes/">GraphFrames</a> is a package for <a href="https://github.com/apache/spark">Apache Spark</a> that provides a DataFrame-based API for working with graphs. Functionality includes motif finding and common graph algorithms, such as <a href="https://en.wikipedia.org/wiki/PageRank">PageRank</a> and <a href="https://en.wikipedia.org/wiki/Breadth-first_search">Breadth-first</a> search.</p>
<p><strong>Objective:</strong> Can we refactor the graph backend of the BioPlex data package using <code>graphframes</code>?</p>
<p><strong>Proposed methods:</strong> The graph backend of the BioPlex data package is currently based on the Bioconductor <a href="https://bioconductor.org/packages/graph">graph</a> package. The function <code>bioplex2graph</code> turns an ordinary <code>data.frame</code> storing the BioPlex PPI data into a <code>graphNEL</code> object, where bait and prey relationships are represented by directed edges from bait to prey. Data for individual nodes/proteins is stored in the <code>nodeData</code> slot, and data for the edges/interactions is stored in the <code>edgeData</code> slot. We might also want to store graph-level metadata such as cell line, network version, and PMID of the primary publication associated with a PPI network, among others. The function <code>bioplex2graph</code> will need to be adapted to be based on <code>GraphFrames</code> instead.</p>
<p><strong>What a successful result would look like:</strong> Pull request to the <a href="https://github.com/ccb-hms/BioPlex">BioPlex</a> github repository on a new branch (named <em>graphframes</em>). Pull requests will be reviewed and discussed. Contributions will be acknowledged.</p>
<p><strong>Potential follow-up work:</strong> Several functions of the BioPlex package add data via the <a href="https://bioconductor.org/packages/graph">graph</a> API to the <code>nodeData</code> and the <code>edgeData</code> of a BioPlex graph (typically obtained via <code>bioplex2graph</code>). This includes (i) the <code>annotatePFAM</code> function, which adds PFAM protein domain annotations to the nodes, and (ii) the <code>mapSummarizedExperimentOntoGraph</code> function, which allows to transfer <code>assay</code> data and/or <code>rowData</code> annotations from the rows of a <code>SummarizedExperiment</code> to the node data of a graph. The implementation of these functions will need to be adapted to make use of the <code>GraphFrames</code> API instead.</p>
</div>
<div id="project-4-shiny-graph-viewer-for-exploration-of-node-and-edge-data" class="section level2">
<h2 class="hasAnchor">
<a href="#project-4-shiny-graph-viewer-for-exploration-of-node-and-edge-data" class="anchor"></a>Project 4: Shiny graph viewer for exploration of node and edge data</h2>
<p><strong>Background:</strong> A variety of data and annotations are available for the nodes (= proteins) and the edges (= PPIs) of the BioPlex networks. As illustrated in the <a href="https://ccb-hms.github.io/BioPlex/articles/BioPlex.html">BioPlex data retrieval vignette</a>, this includes (i) for the nodes: transcript and protein abundance, results of differential expression analysis, disease association scores, etc., and (ii) for the edges: confidence scores, interaction probabilities, etc. On the other hand, graph algorithms such as maximum scoring subnetwork analysis, potentially combined with gene set enrichment analysis, typically produce subnetwork/pathway-sized graphs of interest prompting further interactive exploration.</p>
<p><strong>Objective</strong>: Setting up an R/Shiny graph viewer that allows exploration of data and annotations for nodes and edges of the networks.</p>
<p><strong>Proposed methods:</strong> The <a href="https://github.com/ccb-hms/GraphViewer">GraphViewer</a> github repository contains a scaffold implementation based on <a href="https://cran.r-project.org/web/packages/ggnetwork/index.html">ggnetwork</a>. Alternatives implementing a <a href="https://cran.r-project.org/web/packages/ggnetwork/index.html">ggplot2</a> approach to the visualization of graphs and networks such as <a href="https://cran.r-project.org/web/packages/ggraph/index.html">ggraph</a> exist. The app is supposed to take subnetwork/pathway-sized graphs of interest as input, and is expected to support overlay of node attributes and edge attributes using ggplot2 grammar. The UI should be minimal. A drop-down for the node data attribute to display on the graph, and a drop-down to display the edge data attribute to display on a graph. Upload of a serialized graph object by the user should be supported.</p>
<p><strong>What a successful result would look like</strong>: Pull request to the <a href="https://github.com/ccb-hms/GraphViewer">GraphViewer</a> github repository. Pull requests will be reviewed and discussed. Contributions will be acknowledged.</p>
<p><strong>Potential follow-up work:</strong> Integration of the graph viewer as a new panel type for <a href="https://bioconductor.org/packages/iSEE">iSEE</a>, which takes a panel-based approach to visualizing assay data, column data, and row data of a <code>SummarizedExperiment</code>. Extensions for individual panels for visualizing node data and edge data of a graph could be modeled after the <code>RowDataPlot</code> and <code>ColumnDataPlot</code> classes.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kevin Rue-Albrecht, Vince Carey, Johannes Rainer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
