<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Systematic analysis of published microbial signatures from BugSigDB • BiocChallenges</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../../bootstrap-toc.css">
<script src="../../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../../pkgdown.css" rel="stylesheet">
<script src="../../pkgdown.js"></script><link href="../../extra.css" rel="stylesheet">
<meta property="og:title" content="Systematic analysis of published microbial signatures from BugSigDB">
<meta property="og:description" content="BiocChallenges">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../../index.html">BiocChallenges</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../../articles/BiocChallenges.html">Get started</a>
</li>
<li>
  <a href="../../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../../articles/challenges/biocchallenges_functionality.html">Contribute new functionality to BiocChallenges</a>
    </li>
    <li>
      <a href="../../articles/challenges/bioplex.html">Integrative analysis of proteome-scale protein-protein interaction networks from the BioPlex project</a>
    </li>
    <li>
      <a href="../../articles/challenges/bugsigdb.html">Systematic analysis of published microbial signatures from BugSigDB</a>
    </li>
    <li>
      <a href="../../articles/challenges/build_failure_tracking.html">Facilitating diagnosis of Bioconductor package build failures</a>
    </li>
    <li>
      <a href="../../articles/challenges/instrumentation.html">Instrumentation for genomic computing with Bioconductor</a>
    </li>
    <li>
      <a href="../../articles/challenges/isee_transcripts.html">Unlocking iSEE for transcript-level visualization</a>
    </li>
    <li>
      <a href="../../articles/challenges/single_cell_gallery.html">Contribute to the SingleCellExperiment Gallery</a>
    </li>
    <li class="divider">
    <li>
      <a href="../../articles/index.html">More...</a>
    </li>
  </ul>
</li>
<li>
  <a href="../../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kevinrue/BiocChallenges/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="bugsigdb_files/header-attrs-2.9/header-attrs.js"></script><link href="bugsigdb_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="bugsigdb_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Systematic analysis of published microbial signatures from BugSigDB</h1>
                        <h4 class="author">Ludwig Geistlinger</h4>
            <address class="author_afil">
      Center for Computational Biomedicine, Harvard Medical School<br><a class="author_email" href="mailto:#"></a><a href="mailto:ludwig_geistlinger@hms.harvard.edu" class="email">ludwig_geistlinger@hms.harvard.edu</a>
      </address>
                              <h4 class="author">Levi Waldron</h4>
            <address class="author_afil">
      School of Public Health, City University of New York<br><small class="dont-index">Source: <a href="https://github.com/kevinrue/BiocChallenges/blob/master/vignettes/challenges/bugsigdb.Rmd"><code>vignettes/challenges/bugsigdb.Rmd</code></a></small>
      <div class="hidden name"><code>bugsigdb.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p><a href="https://bugsigdb.org">BugSigDB</a> is a manually curated database of microbial signatures from published differential abundance studies, providing standardized data on geography, health outcomes, host body sites, and experimental, epidemiological, and statistical methods using controlled vocabulary. To date, BugSigDB provides more than 2,000 signatures from over 500 published studies, allowing systematic assessment of microbiome abundance changes within and across experimental conditions and body sites. The bugsigdbr package (<a href="https://github.com/waldronlab/bugsigdbr" class="uri">https://github.com/waldronlab/bugsigdbr</a>, submitted to Bioconductor) implements access to BugSigDB from within R/Bioconductor. This includes import of BugSigDB data into R/Bioconductor, utilities for extracting microbe signatures, and export of the extracted signatures to plain text files in standard file formats such as GMT. The goal of this challenge is to introduce the BugSigDB database and package to the community, and work together on several analysis and programming challenges around the data including: (a) identification of body site-specific signatures from healthy samples, (b) efficient calculation of similarity measures between signatures across the whole database or specific subsets of it, (c) automatic identification of candidate papers for curation based on recently proposed text mining approaches, (d) ontology-based queries to the database using controlled vocabulary for experimental factors and body sites, and (e) inference of abundance changes along the taxonomic hierarchy using phylogenetic approaches such as ancestral state reconstruction.</p>
    </div>
    
<div id="difficulty" class="section level2">
<h2 class="hasAnchor">
<a href="#difficulty" class="anchor"></a>Difficulty</h2>
<p>Advanced.</p>
</div>
<div id="challenge-leaders" class="section level2">
<h2 class="hasAnchor">
<a href="#challenge-leaders" class="anchor"></a>Challenge leader(s)</h2>
<ul>
<li>Ludwig Geistlinger - <i class="fab fa-github"></i> <a href="https://github.com/lgeistlinger">lgeistlinger</a>
</li>
<li>Levi Waldron - <i class="fab fa-github"></i> <a href="https://github.com/lwaldron">lwaldron</a>
</li>
</ul>
</div>
<div id="date-added" class="section level2">
<h2 class="hasAnchor">
<a href="#date-added" class="anchor"></a>Date added</h2>
<p>This challenge was created on August 05, 2021</p>
</div>
<div id="topics" class="section level2">
<h2 class="hasAnchor">
<a href="#topics" class="anchor"></a>Topics</h2>
<ul>
<li><code>BugSigDB</code></li>
</ul>
</div>
<div id="duration" class="section level2">
<h2 class="hasAnchor">
<a href="#duration" class="anchor"></a>Duration</h2>
<p>A minimum of TBD is recommended for this challenge.</p>
</div>
<div id="repositories" class="section level2">
<h2 class="hasAnchor">
<a href="#repositories" class="anchor"></a>Repositories</h2>
<p>A list of GitHub repositories that participate in this challenge.</p>
<pre><code>## Warning in .cache_github_repo(repo): Querying github

## Warning in .cache_github_repo(repo): Querying github</code></pre>
<table class="table">
<thead><tr class="header">
<th align="left">repository</th>
<th align="right">stargazers</th>
<th align="left">latest.push</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">
<i class="fab fa-github"></i> <a href="https://github.com/waldronlab/bugsigdbr">waldronlab/bugsigdbr</a>
</td>
<td align="right">0</td>
<td align="left">2021-07-21 14:45:43</td>
</tr>
<tr class="even">
<td align="left">
<i class="fab fa-github"></i> <a href="https://github.com/waldronlab/BugSigDBStats">waldronlab/BugSigDBStats</a>
</td>
<td align="right">0</td>
<td align="left">2021-08-04 05:11:26</td>
</tr>
</tbody>
</table>
</div>
<div id="project-1-addition-of-body-site-typical-signatures" class="section level2">
<h2 class="hasAnchor">
<a href="#project-1-addition-of-body-site-typical-signatures" class="anchor"></a>Project 1: Addition of body site-typical signatures</h2>
<p><strong>Background:</strong> <a href="https://bugsigdb.org">bugsigdb.org</a> provides signatures of differentially abundant microbial taxa that have been reported in the published literature, and is planned for official public release in the near future. One planned feature is to provide users with automated identification of similar signatures, and this can already be done offline with the <code>calcPairwiseOverlaps</code> function in <a href="https://github.com/waldronlab/BugSigDBStats">BugSigDBStats</a>. A useful comparison to have would be with the “typical” microbes that are present in healthy people at each body site, as these might be expected to be depleted in disease states but not enriched. This is not a typical use case of <a href="https://bugsigdb.org">bugsigdb.org</a> but would be a useful addition nonetheless.</p>
<p><strong>Proposed methods:</strong> Using the <a href="https://bioconductor.org/packages/curatedMetagenomicData">curatedMetagenomicData</a> Biocondutor package (&gt;=3.0.0): create a <a href="https://bioconductor.org/packages/TreeSummarizedExperiment">TreeSummarizedExperiment</a> object per body site for all profiles from healthy individuals. See the <code>body_site</code> and <code>disease</code> columns in the <code>sampleMetadata</code> object, and the <code>returnSamples</code> function. Use the <a href="https://rdrr.io/github/FelixErnst/mia/man/splitByRanks.html">mia::splitByRanks</a> function to add genus-level <code>altExps</code> to the <code>SummarizedExperiment</code> objects.</p>
<p>Separately for each body site and for species/genus levels, identify taxa that have relative abundance &gt; 0 in at least 50% of samples. Write these to file as comma-separated NCBI IDs. Also write to file the number of samples in each body site. You can use something simple like <code>writeLines</code> to write out anything that is needed for <a href="https://bugsigdb.org">bugsigdb.org</a> data entry, then the NCBI IDs will be copy-pasted into the signature entry form. Also write the species and genus matrices to file. Upload code and result files to <a href="https://zenodo.org">zenodo.org</a> to get a DOI.</p>
<p>Add these signatures as a Study in <a href="https://bugsigdb.org">bugsigdb.org</a>, including the Zenodo DOI. It can be one Study, an Experiment for each body site, and a Signature for species plus a signature for genus in each body site. Note that comma-separated NCBI IDs can be bulk copy-pasted into the Signature entry form.</p>
<p><strong>What a successful result would look like:</strong> One new study in <a href="https://bugsigdb.org">bugsigdb.org</a> referencing the Zenodo DOI, containing one Experiment per body site in <a href="https://bioconductor.org/packages/curatedMetagenomicData">curatedMetagenomicData</a>, and two signatures (species and genus) per Experiment. Some creativity will be required for figuring out how to fill in Experiment fields in [bugsigdb.org, for example “group 0 name” might be “blank control” with sample size 0, and we might have to allow some new statistical methods to be entered, because this isn’t standard.</p>
<p><strong>Potential follow-up work:</strong> This study would be used in analyses of other BugSigDB signatures, and mentioned in the manuscript in preparation.</p>
</div>
<div id="project-2-fast-conversion-and-analysis-of-signature-similarity" class="section level2">
<h2 class="hasAnchor">
<a href="#project-2-fast-conversion-and-analysis-of-signature-similarity" class="anchor"></a>Project 2: Fast conversion and analysis of signature similarity</h2>
<p><strong>Background:</strong> There are currently more than 2,000 published signatures in BugSigDB, and we would like to do graph-based and clustering analysis based on all pairwise similarities of these signatures, such as by <a href="https://en.wikipedia.org/wiki/Jaccard_index">Jaccard index</a>. Performant methods are needed to calculate such a large distance matrix, visualize similar signatures, and add a small number of new signatures to the comparison without re-computing the entire distance matrix.</p>
<p><strong>Proposed methods:</strong> The <code>calcPairwiseOverlaps</code> function from <a href="https://github.com/waldronlab/BugSigDBStats">BugSigDBStats</a> currently does a reasonably fast job of calculating pairwise overlaps between BugSigDB signatures (see the <a href="http://waldronlab.io/BugSigDBStats/articles/BugSigDBStats.html">vignette</a> for full usage including data import using the <a href="https://github.com/waldronlab/bugsigdbr">bugsigdbr</a> package), returning a long-format <code>data.frame</code> providing a network edge list.</p>
<p><strong>What a successful result would look like:</strong> Pull requests to <a href="https://github.com/waldronlab/BugSigDBStats">BugSigDBStats</a> for 1) calculation of a full distance matrix for ~2,000 signatures in a few seconds, 2) cluster analysis, e.g. k-means clustering and hierarchical clustering.</p>
<p><strong>Potential follow-up work:</strong> Provision of a function using the results of above to immediately compare results of a new differential abundance analysis to the existing signatures in BugSigDB. A shiny app to do the same.</p>
</div>
<div id="project-3-automatic-identification-of-candidate-papers" class="section level2">
<h2 class="hasAnchor">
<a href="#project-3-automatic-identification-of-candidate-papers" class="anchor"></a>Project 3: Automatic identification of candidate papers</h2>
<p><strong>Background:</strong> The ultimate goal of BugSigDB would be to capture the entire literature of human microbiome studies that have reported signatures of differentially abundant microbial taxa in some comparison of different study subjects. One challenge to meeting this goal is finding published studies that meet the basic criteria for inclusion: 1) the study is indexed in PubMed, 2) the study reports one or more microbial signatures of host-associated microbiota, ie lists of microbial taxa found to be differentially abundant between study conditions or groups of study subjects.</p>
<p><strong>Proposed methods:</strong> A recently published tool called ASReview (<a href="https://www.nature.com/articles/s42256-020-00287-7">van de Schoot et al., 2021</a>) applies machine learning to assist with the prioritization of papers for systematic review. ASReview takes as input text (ie titles and abstracts) from search results, along with “true positive” results that have been manually confirmed to be relevant to the review, then predicts which other publications are most likely to be relevant.</p>
<p><a href="https://pypi.org/project/asreview/">ASReview</a> is a command-line tool written in Python, that must be installed and run on the command-line. BugSigDB provides bulk export of its contents, including study PMIDs, titles, and abstracts, that could be used as the “true positive” input data (for example from the <a href="https://bugsigdb.org/Help:Export">Studies export page</a> or from the <a href="https://github.com/waldronlab/bugsigdbr">bugsigdbr</a> R package).</p>
<p>Steps to apply ASReview to this problem would be something like: Find a PMID search term that captures most of the studies already in BugSigDB. This will probably return tens or hundreds of thousands of studies. Doing this precisely would require comparison to the &gt;500 studies already in BugSigDB, but this isn’t critical at least at first, and a general PMID search like <code>(microbiome or microbiota) AND (sequencing OR 16S OR shotgun OR amplicon)</code> (currently returning &gt;33,000 results) could be used as a start. One would then need to figure out how to use this search result as input to ASReview, and how to use the current studies in BugSigDB as positive results input in ASReview. Subsequently, one would use ASReview to produce a ranking of likely relevant studies not yet included in BugSigDB.</p>
<p><strong>What a successful result would look like:</strong> All studies near the top of the ASReview ranked list should be appropriate for entry into <a href="https://bugsigdb.org">BugSigDB</a>. The process should be easy to automate and update. On updates, studies already in BugSigDB should not be shown, leaving only candidates for new entry.</p>
<p><strong>Potential follow-up work:</strong> This search process could be incorporated into a GitHub Action and the results kept up to date in a public location, and provide a go-to list for studies that can be entered into BugSigDB. People wanting to do a systematic review of studies reporting differential microbial abundance for one particular health condition or exposure could make minor modifications to the process to assist in identifying studies for their review.</p>
</div>
<div id="project-4-ontology-based-queries-for-experimental-factors-and-body-sites" class="section level2">
<h2 class="hasAnchor">
<a href="#project-4-ontology-based-queries-for-experimental-factors-and-body-sites" class="anchor"></a>Project 4: Ontology-based queries for experimental factors and body sites</h2>
<p><strong>Background:</strong> The Semantic MediaWiki curation interface at bugsigdb.org enforces metadata annotation of signatures to follow established ontologies such as the <a href="https://www.ebi.ac.uk/efo/">Experimental Factor Ontology (EFO)</a> for condition, and the <a href="https://www.ebi.ac.uk/ols/ontologies/uberon">Uber-Anatomy Ontology (UBERON)</a> for body site. The <a href="https://github.com/waldronlab/bugsigdbr">bugsigdbr</a> package implements access to BugSigDB from within R/Bioconductor. This includes import of BugSigDB data via the <code>importBugSigDB</code> function into an ordinary <code>data.frame</code> from which subsets of interests can be obtained. Such subsets can eg be obtained for signatures associated with certain experimental factors or specific body sites of interest.</p>
<p><strong>Objective:</strong> Support ontology-based subsetting of BugSigDB signatures.</p>
<p><strong>Proposed methods:</strong> The <a href="https://cran.r-project.org/web/packages/ontologyIndex/index.html">ontologyIndex</a> package implements functions for reading and querying ontologies in R. This includes the <code>get_ontology</code> function for reading ontologies from files in <a href="https://owlcollab.github.io/oboformat/doc/GO.format.obo-1_4.html">OBO format</a>. The OBO file for EFO is available <a href="https://www.ebi.ac.uk/efo/efo.obo">here</a> and the OBO file for UBERON is available <a href="http://ontologies.berkeleybop.org/uberon.obo">here</a>. Subsetting BugSigDB signatures by an EFO term will then involve subsetting the <code>Condition</code> column to all descendants of that term in the EFO ontology and that are present in the <code>Condition</code> column. And analogously, subsetting by an UBERON term will then involve subsetting the <code>Body site</code> column to all descendants of that term in the UBERON ontology and that are present in the <code>Body site</code> column.</p>
<p><strong>What a successful result would look like:</strong> Pull request to the <a href="https://github.com/waldronlab/bugsigdbr">bugsigdbr</a> github repository on a new branch (named <em>ontoquery</em>). Pull requests will be reviewed and discussed. Contributions will be acknowledged.</p>
<p><strong>Potential follow-up work:</strong> Discussion of how to implement high-level queries also for other columns of interest such as <code>Location of subjects</code>, <code>Host species</code>, and <code>Statistical test</code>.</p>
</div>
<div id="project-5-inference-of-abundance-changes-via-ancestral-state-reconstruction" class="section level2">
<h2 class="hasAnchor">
<a href="#project-5-inference-of-abundance-changes-via-ancestral-state-reconstruction" class="anchor"></a>Project 5: Inference of abundance changes via ancestral state reconstruction</h2>
<p><strong>Background:</strong> Differential abundance studies typically report signatures resolved to the genus level (16S rRNA sequencing) or species level (whole-metagenome sequencing). In some cases, authors also make statements about differential abundance on a higher taxonomic rank for cases where supposedly all (at least all investigated) taxa of a lower taxonomic rank display differential abundance. An example would be if both known species of the genus <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&amp;id=1911312">Gabonibacter</a>, ie <em>Gabonibacter massiliensis</em> and <em>Gabonibacter timonensis</em>, would be found with increased abundance in a certain condition, authors would sometimes conclude that the genus <em>Gabonibacter</em> is overall found with increased abundance, resulting in mixed signatures of different taxonomic ranks.</p>
<p><a href="https://en.wikipedia.org/wiki/Ancestral_reconstruction">Ancestral state reconstruction (ASR)</a> is a phylogenetic approach for inferring ancestor states from characteristics measured for their descendants. For example, given differential abundance on the species level, this could thus be used to infer differential abundance on the genus level, and further up the taxonomy.</p>
<p><strong>Objective:</strong> Can we apply ASR for harmonization of BugSigDB signatures to a given taxonomic rank?</p>
<p><strong>Proposed method</strong>: Microbial signatures in BugSigDB follow the nomenclature of the <a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI Taxonomy Database</a>, restricted to microbial clades profiled by <a href="https://github.com/biobakery/MetaPhlAn">MetaPhlAn3</a>. The phylogenetic tree for MetaPhlan3 species in <a href="https://en.wikipedia.org/wiki/Newick_format">Newick format</a> is available <a href="https://raw.githubusercontent.com/biobakery/MetaPhlAn/master/metaphlan/utils/mpa_v30_CHOCOPhlAn_201901_species_tree.nwk">here</a>. The <code>ace</code> function from the <a href="https://cran.r-project.org/web/packages/ape/index.html">ape</a> package provides a standard implementation of ancestral state reconstruction based on maximum likelihood estimation for discrete characters (here: “UP” or “DOWN” to indicate increased or decreased abundance.)</p>
<p><strong>What a successful result would look like:</strong> Pull request to the <a href="https://github.com/waldronlab/BugSigDBStats">BugSigDBStats</a> github repository outlining the approach in a separate ancestral state reconstruction vignette (<code>.Rmd</code> file in the <code>vignettes</code> folder). Pull requests will be reviewed and discussed. Accepted contributions will be acknowledged.</p>
<p><strong>Potential follow-up work:</strong> If shown to be a feasible approach, incorporation into <code>bugsigdbr::getSignatures</code> should be considered. The logical argument <code>exact.tax.level</code> currently harmonizes signatures by only including taxa given at the indicated <code>tax.level</code> (<code>exact.tax.level = TRUE</code>), or extracts a more general <code>tax.level</code> for microbes given at a more specific taxonomic level by simply cutting the tree at the desired <code>tax.level</code> (<code>exact.tax.level = FALSE</code>). Alternatively, simple majority votes or estimation via ASR could be provided as additional options for the <code>exact.tax.level</code> argument.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kevin Rue-Albrecht.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
